Python/ceval.c

```diff
@@ -2248,6 +2249,34 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
             NOTRACE_DISPATCH();
         }
 
+        TARGET(BINARY_SUBSCR_ASCII_INT) {
+            assert(cframe.use_tracing == 0);
+            PyObject *sub = TOP();
+            PyObject *unicode = SECOND();
+            DEOPT_IF(!PyLong_CheckExact(sub), BINARY_SUBSCR);
+            DEOPT_IF(!PyUnicode_CheckExact(unicode), BINARY_SUBSCR);
+            DEOPT_IF(!PyUnicode_IS_ASCII(unicode), BINARY_SUBSCR);
+
+            // Deopt unless 0 <= sub < ascii->length
+            PyASCIIObject *ascii = _PyASCIIObject_CAST(unicode);
+            Py_ssize_t signed_magnitude = Py_SIZE(sub);
+            DEOPT_IF(((size_t)signed_magnitude) > 1, BINARY_SUBSCR);
+            assert(((PyLongObject *)_PyLong_GetZero())->ob_digit[0] == 0);
+            Py_ssize_t index = ((PyLongObject*)sub)->ob_digit[0];
+            DEOPT_IF(index >= ascii->length, BINARY_SUBSCR);
+            STAT_INC(BINARY_SUBSCR, hit);
+            Py_UCS1 ch = *((Py_UCS1*)(ascii + 1) + index);
```

Comment:
This is a bit obscure. Is there a pre-existing macro for getting the first character in an ascii string?